Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1736,6 +1746,343 @@ RefPtr<StyleValue> Parser::parse_image_value(ParsingContext const& context, Styl
     return {};
 }
 
+static inline bool is_background_repeat(StyleValue const& value)
+{
+    switch (value.to_identifier()) {
+    case ValueID::NoRepeat:
+    case ValueID::Repeat:
+    case ValueID::RepeatX:
+    case ValueID::RepeatY:
+    case ValueID::Round:
+    case ValueID::Space:
+        return true;
+    default:
+        return false;
+    }
+}
+
+static inline bool is_background_image(StyleValue const& value)
```

Comment:
Isn't this rather a `is_image()`?

Reply:
I'm genuinely not sure. `none` might be a valid value for other places that accept an image, but it might not be. [The `background-image` spec](https://www.w3.org/TR/css-backgrounds-3/#background-image) says:
> Value: `<bg-image>#`
> ...
> Where
> `<bg-image> = <image> | none`

which implies that `<image>` and `none` are different things generally. I don't know. For now, I think it might be a trap to call it `is_image()` in case it's not true.

Reply:
Better to stay with spec-language then.
I was just a bit confused because the function itself doesn't do anything that looks specifically 'background'-ish.

Reply:
Yeah that's fair. And it started out as a lambda inside `parse_background_value()`, like the other ones, so I didn't think to change the name when I moved it.

There's sadly no way to check what values can be in CSS globally - it's all defined per property.